### PR TITLE
feat: Implement create wallet menu

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements create wallet menu on the X1 vault. This menu can be reached by navigating the correct option from the main menu (only if maximum wallets are not created already)

### CODE REVIEW

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]
Idea: Auto increment the count by making the `NUMBER_OF_OPTIONS_NEW_WALLET` as an enum of `create_wallet_options_e`.

**Comment by kartikCypherock** [comment]
> Idea: Auto increment the count by making the `NUMBER_OF_OPTIONS_NEW_WALLET` as the last enum of `create_wallet_options_e`.

While that's a good idea, but anyway 2 macros/enums are required to handle this as we want to keep constant texts in one location.

File: src/menu/create_wallet_menu.c [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static void create_wallet_menu_initialize(engine_ctx_t *ctx,
+                                          const void *data_ptr) {
+  handle_core_errors();
```
[the remark this answers is not in the document]
Comment (ujjwal-cyph) [reply src/menu/create_wallet_menu.c]: thanks. agree, we can take an opinion on the behavior change.

**Review by ujjwal-cyph — APPROVED** [review]

File: src/menu/create_wallet_menu.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+ */
+static void create_wallet_menu_handler(engine_ctx_t *ctx,
+                                       ui_event_t ui_event,
+                                       const void *data_ptr);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+const flow_step_t create_wallet_step = {
+    .step_init_cb = create_wallet_menu_initialize,
+    .p0_cb = NULL,
```
Comment (amanCypherock) [inline src/menu/create_wallet_menu.c:115]: p0 handler missing

Comment (amanCypherock) [reply src/menu/create_wallet_menu.c:115]: what is the intended behavior of P0 here anyway?

Comment (kartikCypherock) [reply src/menu/create_wallet_menu.c]: I think on menu, P0 event is not expected.. so I have increased P0 timeout to `INFINITE_WAIT_TIMEOUT`.

As a precautionary step (in the case P0 ever occurs), I have added the callback to clear p0  event

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**